File: app/src/main/java/com/github/premnirmal/ticker/network/Query.java
package com.github.premnirmal.ticker.network;

import android.os.Parcel;
import android.os.Parcelable;

import com.github.premnirmal.ticker.network.historicaldata.Results;


public class Query implements Parcelable{

    public int count;
    public Results results;
    public int executionTime;
    public int executionStartTime;
    public int executionStopTime;
    public String param;
    public String created;
    public String content;
    public String lang;

    public Query(){

    }

    public Query(Parcel in) {
        count = in.readInt();
        results = in.readParcelable(Results.class.getClassLoader());
        executionTime = in.readInt();
        executionStartTime = in.readInt();
        executionStopTime = in.readInt();
        param = in.readString();
        created = in.readString();
        content = in.readString();
        lang = in.readString();
    }

    @Override
    public int describeContents() {
        return 0;
    }

    public static final Creator<Query> CREATOR = new Creator<Query>() {
        public Query createFromParcel(Parcel in) {
            return new Query(in);
        }

        public Query[] newArray(int size) {
        return new Query[size];
        }
    };

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeInt(count);
        dest.writeParcelable(results, flags);
        dest.writeInt(executionTime);
        dest.writeInt(executionStartTime);
        dest.writeInt(executionStopTime);
        dest.writeString(param);
        dest.writeString(created);
        dest.writeString(content);
        dest.writeString(lang);
    }


}

File: app/src/main/kotlin/com/github/premnirmal/ticker/network/data/historicaldata/Query.kt
package com.github.premnirmal.ticker.network.data.historicaldata

import android.os.Parcel
import android.os.Parcelable
import android.os.Parcelable.Creator
import com.google.gson.annotations.SerializedName

/**
 * Created by premnirmal on 3/30/17.
 */
class Query : Parcelable {

  companion object {

    const val FIELD_COUNT = "count"
    const val FIELD_RESULTS = "results"
    const val FIELD_CREATED = "created"
    const val FIELD_LANG = "lang"

    val CREATOR: Creator<Query> = object : Creator<Query> {
      override fun createFromParcel(`in`: Parcel): Query {
        return Query(`in`)
      }

      override fun newArray(size: Int): Array<Query> {
        return newArray(size)
      }
    }
  }

  @SerializedName(FIELD_COUNT) var mCount: Int = 0
  @SerializedName(FIELD_RESULTS) var mResult: History? = null
  @SerializedName(FIELD_CREATED) var mCreated: String? = null
  @SerializedName(FIELD_LANG) var mLang: String? = null

  constructor()

  constructor(`in`: Parcel) {
    mCount = `in`.readInt()
    mResult = `in`.readParcelable<History>(History::class.java.classLoader)
    mCreated = `in`.readString()
    mLang = `in`.readString()
  }

  override fun describeContents(): Int {
    return 0
  }

  override fun writeToParcel(dest: Parcel, flags: Int) {
    dest.writeInt(mCount)
    dest.writeParcelable(mResult, flags)
    dest.writeString(mCreated)
    dest.writeString(mLang)
  }
}